cmake/FindCybSDK.cmake

```diff
@@ -1,86 +1,67 @@
-    set(CybSDK_FOUND FALSE)
-    unset(_DIR)
-  endif()
-endmacro()
+if(NOT DEFINED ENV{CybSDK_INCLUDE_DIRS})
+  message( FATAL_ERROR "Environment variable {CybSDK_INCLUDE_DIRS} is not defined: $ENV{CybSDK_INCLUDE_DIRS}" )
+else()
+  message("Environment variable {CybSDK_INCLUDE_DIRS}: $ENV{CybSDK_INCLUDE_DIRS}" )
+endif()
+##### Find CybSDK #####
 
-# Check Files Macro
-macro(CHECK_FILES _FILES _DIR)
-  set(_MISSING_FILES)
-  foreach(_FILE ${${_FILES}})
-    if(NOT EXISTS "${_FILE}")
-      get_filename_component(_FILE ${_FILE} NAME)
-      set(_MISSING_FILES "${_MISSING_FILES}${_FILE}, ")
-    endif()
-  endforeach()
-  if(_MISSING_FILES)
-    message(WARNING "In directory \"${${_DIR}}\" not found files: ${_MISSING_FILES}")
-    set(CybSDK_FOUND FALSE)
-    unset(_FILES)
-  endif()
-endmacro()
+find_path(CybSDK_INCLUDE_DIR NAMES CVirt.h CVirtDevice.h HINTS $ENV{CybSDK_INCLUDE_DIRS})
 
-# Target Platform
-set(TARGET_PLATFORM)
-if(NOT CMAKE_CL_64)
-  set(TARGET_PLATFORM x86)
+if(WIN32)
+  find_library(CybSDK_LIBRARY NAMES CybSDK HINTS ${CybSDK_DIR})
 else()
-  set(TARGET_PLATFORM x64)
+  message(FATAL_ERROR  "The Sense Glove SDK is not supported for this operating system.")
 endif()
 
-##### Find CybSDK #####
+find_package_handle_standard_args(CybSDK_LIBRARY DEFAULT_MSG CybSDK_LIBRARIES
+                                                      CybSDK_INCLUDE_DIR)
 
-# Found
-set(CybSDK_FOUND TRUE)
-if(MSVC_VERSION LESS 1700)
-  message(WARNING "CybSDK supported Visual Studio 2012 or later.")
-  set(CybSDK_FOUND FALSE)
-endif()
 
-# Root Directoty
-set(CybSDK_DIR)
-if(CybSDK_FOUND)
-  set(CybSDK_DIR $ENV{CybSDK_DIR} CACHE PATH "CybSDK Path." FORCE)
-  check_dir(CybSDK_DIR)
+if(CybSDK_INCLUDE_DIR AND CybSDK_LIBRARY)
+	set(CybSDK_FOUND TRUE)
+else()
+	set(CybSDK_FOUND FALSE)
 endif()
 
-# Include Directories
-set(CybSDK_INCLUDE_DIRS)
-if(CybSDK_FOUND)
-  set(CybSDK_INCLUDE_DIRS ${CybSDK_DIR}/include)
-  check_dir(CybSDK_INCLUDE_DIRS)
-endif()
+set(CybSDK_LIBRARIES ${CybSDK_LIBRARY})
 
-# Library Directories
-set(CybSDK_LIBRARY_DIRS)
-if(CybSDK_FOUND)
-  set(CybSDK_LIBRARY_DIRS ${CybSDK_DIR}/Plugins/${TARGET_PLATFORM})
-  check_dir(CybSDK_LIBRARY_DIRS)
-endif()
+message(STATUS "CybSDK_FOUND: ${CybSDK_FOUND}")
+message(STATUS "CybSDK_LIBRARIES: ${CybSDK_LIBRARIES}")
+message(STATUS "CybSDK_INCLUDE_DIR: ${CybSDK_INCLUDE_DIR}")
 
-# Dependencies
-set(CybSDK_LIBRARIES)
-if(CybSDK_FOUND)
-  set(CybSDK_LIBRARIES ${CybSDK_LIBRARY_DIRS}/CybSDK.dll)
-  check_files(CybSDK_LIBRARIES CybSDK_LIBRARY_DIRS)
-endif()
+#set(CybSDK_FOUND TRUE)
+#if(NOT CybSDK )
+#  message(FATAL_ERROR "Sense Glove not found.")
+#  set(CybSDK_FOUND FALSE)
+#endif()
+
+# Include Directories
+#set(CybSDK_INCLUDE_DIRS)
+#if(CybSDK_FOUND)
+#  set(CybSDK_INCLUDE_DIRS $ENV{CybSDK_DIR}/Core/SGCoreCpp/incl)
+#endif()
+
+#if(NOT (EXISTS CybSDK_INCLUDE_DIRS))
+#  message(FATAL_ERROR "CybSDK_INCLUDE_DIRS not exists: ${CybSDK_INCLUDE_DIRS}")
+#endif()
```

Comment:
Please avoid leaving commented code around. It may make the user wonder if those lines are actually useful or not. Same comment for the other commented code.